ext.h

```diff
@@ -33,3 +33,44 @@ static int secp256k1_ext_reencode_pubkey(
 	unsigned int flag = (outlen == 33) ? SECP256K1_EC_COMPRESSED : SECP256K1_EC_UNCOMPRESSED;
 	return secp256k1_ec_pubkey_serialize(ctx, out, &outlen, &pubkey, flag);
 }
+
+// secp256k1_ext_scalar_mul multiplies a point by a scalar in constant time.
+//
+// Returns: 1: multiplication was successful
+//          0: scalar was invalid (zero or overflow)
+// Args:    ctx:      pointer to a context object (cannot be NULL)
+//  Out:    point:    the multiplied point (usually secret)
+//  In:     point:    pointer to a 64-byte public point,
+//                    encoded as two 256bit big-endian numbers.
+//          scalar:   a 32-byte scalar with which to multiply the point
+int secp256k1_ext_scalar_mul(const secp256k1_context* ctx, unsigned char *point, const unsigned char *scalar) {
+	int ret = 0;
+	int overflow = 0;
+	secp256k1_fe feX, feY;
+	secp256k1_gej res;
+	secp256k1_ge ge;
+	secp256k1_scalar s;
+	ARG_CHECK(point != NULL);
+	ARG_CHECK(scalar != NULL);
+	(void)ctx;
+
+	// TODO(yperbasis): limit or mod?
+	secp256k1_fe_set_b32_limit(&feX, point);
```

Comment:
`limit` if you know that the input point is valid. But you may want to check the result of this function.

Reply:
The result of the function call was ignored before. https://pkg.go.dev/crypto/elliptic says "The behavior of Add, Double, and ScalarMult when the input is not a point on the curve is undefined."